### PR TITLE
Initial code for merge CSV cloud function.

### PR DESCRIPTION
**Pull request by sylmak**
Initial code for merge CSV cloud function.

Also, refactor github workflows.

### CODE REVIEW

File: cloud_functions/climateiq_merge_output_cf/main.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from cloudevents import http
+import functions_framework
+
+
+@functions_framework.cloud_event
+def merge_predictions(cloud_event: http.CloudEvent) -> None:
```
Comment (skeshive) [inline cloud_functions/climateiq_merge_output_cf/main.py]: nit: merge_scenario_predictions

File: .github/workflows/climateiq_merge_output_cf.yml
Comment (skeshive) [inline .github/workflows/climateiq_merge_output_cf.yml:1]: hmm can this config be shared between this new fucntion and the existing cloud function?

Comment (sylmak) [reply .github/workflows/climateiq_merge_output_cf.yml:1]: If want to trigger them separately (i.e., only run the relevant actions when files in each respective function are updated) then that's not possible.

Comment (skeshive) [reply .github/workflows/climateiq_merge_output_cf.yml:1]: hmm the submitted code should pass all the checks and the checks are pretty quick to run,  I would prefer a single file to reduce overhead for USL

Comment (sylmak) [reply .github/workflows/climateiq_merge_output_cf.yml:1]: That would look like two jobs in the same file, but the jobs would be identical to now, and I don't think having a single long file is significantly less complex than two shorter files. (The requirements for both functions are different, so we can't run all the actions in the same job.)

**Review by skeshive — APPROVED** [review]

**Pull request merged**